### PR TITLE
[1.7.x] Update to mariadb client 2.4.0

### LINKED ISSUE #2835: Upgrade MariaDB driver to v2.4.0
Apparently, in this release, there's a fix for an issue re: TLS certs on PCF.

This also needs to be backport'd to 1.7.x.

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #2835

### CODE REVIEW

File: spring-cloud-dataflow-dependencies/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
 			<dependency>
 				<groupId>org.mariadb.jdbc</groupId>
 				<artifactId>mariadb-java-client</artifactId>
-				<version>2.2.5</version>
+				<version>${mariadb-java-client.version}</version>
```
Comment (jvalkeal) [inline spring-cloud-dataflow-dependencies/pom.xml]: We can't use `mariadb-java-client.version` here as it's not defined in parent pom.

File: spring-cloud-dataflow-server-core/pom.xml [hunk 2]
```diff
@@ -147,6 +147,7 @@
 		<dependency>
 			<groupId>org.mariadb.jdbc</groupId>
 			<artifactId>mariadb-java-client</artifactId>
+            <version>${mariadb-java-client.version}</version>
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/pom.xml]: Version would already come from parent pom's dep management?

**Comment by jvalkeal** [comment]
I was too quick to merge that other PR as it had same issues. This PR should also follow fixes I did in PR #2846

**Comment by markpollack** [comment]
Thanks, I always seem not to be successful `maven pom` surgeon. :(

File: spring-cloud-dataflow-server-core/pom.xml (hunk 2 above)
Comment (markpollack) [reply spring-cloud-dataflow-server-core/pom.xml]: I just try to be explicit, can test.

File: spring-cloud-dataflow-dependencies/pom.xml (hunk 1 above)
Comment (markpollack) [reply spring-cloud-dataflow-dependencies/pom.xml]: That sort of sucks, yet it is building fine.  While I don't think we should change the parent pom here in the 1.7.x branch, maybe we should not have `spring-cloud-dependencies-parent` as the parent for `spring-cloud-dataflow-dependencies` in the master branch for 2.0?

File: spring-cloud-dataflow-server-core/pom.xml (hunk 2 above)
Comment (markpollack) [reply spring-cloud-dataflow-server-core/pom.xml]: seems to work w/o it, so removing it.

**Comment by markpollack** [comment]
closing in favor of https://github.com/spring-cloud/spring-cloud-dataflow/pull/2865

**Pull request closed without merging**